woocommerce/wc-admin
# Make src/Features/* internal except for Features.php, Onboarding.php and TransientNotices.php

Fixes #8274

This PR moves /Features/* to `src-internal` except for `Features.php`, `Onboarding.php` and `TransientNotices.php`.

### Detailed test instructions:

Smoke test WooCommerce Home, analytics pages, marketing page or at least include the following tests:

**ActivityPanels.php**

- Go to WooCommerce Home
- Smoke test the ActivityPanels

**Analytics.php**

- Go to **WooCommerce > Setting > Advanced > Features**
- Confirm that you can toggle **Analytics** feature setting
- Check Analytics report pages are shown properly without network errors.

**Coupons.php** & **CouponsMovedTrait.php** & **Marketing.php**

- Go to **WooCommerce > Coupons**
- Should see the "Coupon management has moved!" message
- Click the "remove" button
- **WooCommerce > Coupons** should not be displayed.
- Go to **Marketing > Coupons**
- Should see the "RecommendedExtensions" and "KnowledgeBase" panels properly.

**CustomerEffortScoreTracks.php**

- Make sure tracking is enabled in settings: 
```
/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com
```
- Delete the option `woocommerce_ces_shown_for_actions` to make sure CES prompt triggers when updating settings.
- Enable the logging of Tracks events to your browser dev console:
```
localStorage.setItem( 'debug', 'wc-admin:tracks' );
```

Testing product categories:
- Go to Products > Categories.
- Add a new category.
- Observe CES prompt "How easy was it to add a product category?" is displayed.

Testing product tags:
- Go to Products > Tags.
- Add a new tag.
- Observe CES prompt "How easy was it to add a product tag?" is displayed.


**Homescreen.php** & **RemoteInboxNotifications.php**

- Go to WooCommerce Home
- Confirm the page can be loaded without any php errors
- Confirm inbox notes are displayed

**MobileAppBanner.php**

This might be hard to test. But this is [a simple script](https://github.com/woocommerce/woocommerce-admin/blob/d9ecfe5644c6bb93cbf82b014cf8521f816c4a64/src-internal/Admin/Features/MobileAppBanner.php). I think if this can be loaded by features.php, it should work properly.

see https://github.com/woocommerce/woocommerce-admin/pull/5037

**Settings.php**

- Set `settings` to true in `config/plugin.json`
- `npm run build:feature-config`
- Go to **WooCommerce > Setting > Advanced > Features**
- Confirm that you can see **Settings** feature setting

**ShippingLabelBanner.php** & **ShippingLabelBannerDisplayRules.php**

reference: https://github.com/woocommerce/woocommerce-admin/pull/3955

- Start with a fresh JN site
- Set Store located in US.
- Install and Connect Jetpack
- Create an order with a shippable product.
- Go to that order's page
- Press the create shipping label button.
- The WooCommerce Services create shipping label modal should pop up, and a new label should be able to be purchased.

**WcPayWelcomePage.php**

- Use a fresh site without woocommerce payments
- Edit `should_add_the_menu` to return `true` in `src/Features/WcPayWelcomePage.php` to simulate treatment experiment  (currently the experiment is disabled for February)
- Should see the "payment" menu item and WCPay welcome page.


no changelog
